torch/distributed/_spmd/distribute.py

```diff
@@ -88,7 +88,7 @@ def _dispatch_with_local_tensors(
 
     def redistribute(arg: object) -> object:
         return (
-            _redistribute_with_local_tensor(arg, *specs[arg])  # type: ignore[index]
+            _redistribute_with_local_tensor(arg, *specs[arg]).wait()  # type: ignore[index]
```

Comment:
will this wait operator get traced into graph?

I assume this is already the latest time wait can be called because we gonna need the args for this compute right away?

Reply:
Yes, after adding the `wait()` call it should be traced into the graph, since this is already where we need to allreduce, it's basically the latest point we need the args, which I think it should be the correct behavior. 

Since `_redistribute_with_local_tensor` always returns AsyncCollectiveTensor. I'm not sure why the previous graph works, probably because we still execute the `tensor_constant` get inlined.